tests/handlers/auth_test.py

```diff
@@ -0,0 +1,358 @@
+    assert r.headers["X-Auth-Request-User"] == "some-user"
+    assert r.headers["X-Auth-Request-Uid"] == "1000"
+    assert r.headers["X-Auth-Request-Groups"] == "admin"
+    assert r.headers["X-Auth-Request-Token"] == token
+    assert r.headers["X-Auth-Request-Token-Ticket"] == ""
+
+
+async def test_authnz_token_success_any(aiohttp_client: TestClient) -> None:
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair, ["test"])
+    app = await create_test_app(keypair)
+    client = await aiohttp_client(app)
+
+    r = await client.get(
+        "/auth",
+        params=[
+            ("capability", "exec:admin"),
+            ("capability", "exec:test"),
+            ("satisfy", "any"),
+        ],
+        headers={"Authorization": f"Bearer {token}"},
+    )
+    assert r.status == 200
+    assert r.headers["X-Auth-Request-Token-Capabilities"] == "exec:test"
+    assert (
+        r.headers["X-Auth-Request-Capabilities-Accepted"]
+        == "exec:admin exec:test"
+    )
+    assert r.headers["X-Auth-Request-Capabilities-Satisfy"] == "any"
+    assert r.headers["X-Auth-Request-Groups"] == "test"
+
+
+async def test_authnz_token_forwarded(aiohttp_client: TestClient) -> None:
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair, ["admin"])
+    app = await create_test_app(keypair)
+    client = await aiohttp_client(app)
+
+    r = await client.get(
+        "/auth",
+        params={"capability": "exec:admin"},
+        headers={
+            "Authorization": "Basic blah",
+            "X-Forwarded-Access-Token": token,
+        },
+    )
+    assert r.status == 200
+    assert r.headers["X-Auth-Request-Email"] == "some-user@example.com"
+
+    r = await client.get(
+        "/auth",
+        params={"capability": "exec:admin"},
+        headers={
+            "Authorization": "Basic blah",
+            "X-Forwarded-Ticket-Id-Token": token,
+        },
+    )
+    assert r.status == 200
+    assert r.headers["X-Auth-Request-Email"] == "some-user@example.com"
+
+
+async def test_authnz_token_basic(aiohttp_client: TestClient) -> None:
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair, ["admin"])
+    app = await create_test_app(keypair)
+    client = await aiohttp_client(app)
+
+    basic = f"{token}:x-oauth-basic".encode()
+    basic_b64 = base64.b64encode(basic).decode()
+    r = await client.get(
+        "/auth",
+        params={"capability": "exec:admin"},
+        headers={"Authorization": f"Basic {basic_b64}"},
+    )
+    assert r.status == 200
+    assert r.headers["X-Auth-Request-Email"] == "some-user@example.com"
+
+    basic = f"x-oauth-basic:{token}".encode()
+    basic_b64 = base64.b64encode(basic).decode()
+    r = await client.get(
+        "/auth",
+        params={"capability": "exec:admin"},
+        headers={"Authorization": f"Basic {basic_b64}"},
+    )
+    assert r.status == 200
+    assert r.headers["X-Auth-Request-Email"] == "some-user@example.com"
```

Comment:
There's also the token is the username and no password case right?

Reply:
Good catch. Added.